src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,229 @@
+import React, {
+  Children,
+  ReactElement,
+  useEffect,
+  useRef,
+  useState,
+} from 'react'
+import classnames from 'classnames'
+interface TooltipProps {
+  label: string
+  position?: 'top' | 'bottom' | 'left' | 'right'
+  className?: string
+  children: ReactElement
+}
+
+const TRIANGLE_SIZE = 5
+const SPACER = 2
+const tooltipID = `tooltip-${Math.floor(Math.random() * 900000) + 100000}`
+
+function isElementInViewport(
+  el: HTMLElement,
+  win = window,
+  docEl = document.documentElement
+): boolean {
+  const rect = el.getBoundingClientRect()
+  return (
+    rect.top >= 0 &&
+    rect.left >= 0 &&
+    rect.bottom <= (win.innerHeight || docEl.clientHeight) &&
+    rect.right <= (win.innerWidth || docEl.clientWidth)
+  )
+}
+
+export const Tooltip = (
+  props: TooltipProps & JSX.IntrinsicElements['span']
+): React.ReactElement => {
+  const { label, position, children, ...spanProps } = props
+  const wrapperRef = useRef<HTMLElement>(null)
+  const triggerElementRef = useRef<HTMLElement>(null)
+  const tooltipBodyRef = useRef<HTMLElement>(null)
+  const tooltipTrigger = React.cloneElement(Children.only(children), {
+    ref: triggerElementRef,
+  })
+  const [isVisible, setVisible] = useState(false)
+
+  const tooltipClasses = classnames('usa-tooltip__body', {
+    'is-set': isVisible,
+    'usa-tooltip__body--top': position === 'top',
+    'usa-tooltip__body--bottom': position === 'bottom',
+    'usa-tooltip__body--right': position === 'right',
+    'usa-tooltip__body--left': position === 'left',
+    'is-visible': isVisible,
+  })
+
+  useEffect(() => {
```

Comment:
I placed the position logic within a useEffect hook because certain calculations like `tooltipBodyWidth` were being generated when the tooltip was offscreen, causing issues with `marginLeft` in `positionLeft`